Error when generating code from tag v1.1.0

I have problems generating java code out of the specs v.1.1.0. OpenAPI generator maven plugin (4.3.1) shows this error:

```
Caused by: java.lang.RuntimeException: Invalid inline schema defined in allOf in 'icarTreatmentProgramEventResource'. Per the OpenApi spec, for this case when a composed schema defines a discriminator, the allOf schemas must use $ref. Change this inline definition to a $ref definition
    at org.openapitools.codegen.DefaultCodegen.getAllOfDescendants (DefaultCodegen.java:2665)
    at org.openapitools.codegen.DefaultCodegen.createDiscriminator (DefaultCodegen.java:2722)
```

--- a/resources/icarTreatmentProgramEventResource.json
+++ b/resources/icarTreatmentProgramEventResource.json
@@ -13,36 +13,22 @@
                         "$ref": "../types/icarDiagnosisType.json"
                     },
                     "description": "Decribes the diagnosis of one or more conditions"
+                },
+                "courses": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "../types/icarMedicineCourseSummaryType.json"
+                    },
+                    "description": "Details the course of treatments at a summary (start and end date) level. The array allows for different medicines/procedures."
+                },
+                "treatments": {
+                    "type": "array",
+                    "items": {
+                        "$ref": "../resources/icarTreatmentEventResource.json"
+                    },
+                    "description": "The list of the treatments (medicines or procedures) applied."
                 }
             }
-        },
-        {
-            "oneOf": [
-                {
-                    "type": "object",
-                    "properties": {
-                        "courses": {
-                            "type": "array",
-                            "items": {
-                                "$ref": "../types/icarMedicineCourseSummaryType.json"
-                            },
-                            "description": "Details the course of treatments at a summary (start and end date) level. The array allows for different medicines/procedures."
-                        }
-                    }
-                },
-                {
-                    "type": "object",
-                    "properties": {
-                        "treatments": {
-                            "type": "array",
-                            "items": {
-                                "$ref": "../resources/icarTreatmentEventResource.json"
-                            },
-                            "description": "The list of the treatments (medicines or procedures) applied."
-                        }
-                    }
-                }
-            ]
         }
     ]
 } 

